src/compiler/checker.ts

```diff
@@ -8150,8 +8150,28 @@ namespace ts {
             return container === declarationContainer;
         }
 
+        function updateReferencesForTypedParameters(node: TypeNode): void {
+            const symbol = getNodeLinks(node).resolvedSymbol;
+            if (symbol) {
+                symbol.hasReference = true;
+            }
+        }
+
+        function updateReferencesForInterfaceImplementations(node: InterfaceDeclaration): void {

```

Comment:
nit, `InterfaceImplementations` is not a language concept, so i would use something already known like HeritageClauseElements instead.